fil_fungible_token/src/token/mod.rs

```diff
@@ -32,39 +32,61 @@ where
 {
     /// Injected blockstore. The blockstore must reference the same underlying storage under Clone
     bs: BS,
-    /// Root of the token state tree
-    state_cid: Cid,
+    /// In-memory cache of the state tree
+    state: TokenState,
 }
 
 impl<BS> Token<BS>
 where
     BS: IpldStore + Clone,
 {
-    /// Instantiate an empty token helper with access to a blockstore and runtime
-    pub fn new(bs: BS, state_cid: Cid) -> Self {
-        Self { bs, state_cid }
+    /// Creates a new token instance using the given blockstore and creates a new empty state tree
+    ///
+    /// Returns a Token handle that can be used to interact with the token state tree and the Cid
+    /// of the state tree root
+    pub fn new(bs: BS) -> Result<(Self, Cid)> {
+        let init_state = TokenState::new(&bs)?;
+        let cid = init_state.save(&bs)?;
```

Comment:
You discarded the Cid return value at every usage. An alternative here is to not save the state here and require a `flush` (after any subsequent modifications).